internal/common/common.go

```diff
@@ -71,6 +71,9 @@ func AllStrings(vs []string, f func(string) bool) bool {
 // GetBootstrapHost return host that was set as bootstrap
 func GetBootstrapHost(cluster *Cluster) *models.Host {
 	for _, host := range cluster.Hosts {
+		if IsSingleNodeCluster(cluster) && cluster.APIVip == "" {
```

Comment:
why? it should have host.Bootstrap ?

Reply:
The bootstrap field at this point is false which results in not returning a host from this method and then errors with `no bootstrap host were found in cluster`.

Reply:
I think it means that we are doing too early

Reply:
for example, there is a possibility that we will have disabled host. Will it work in this case?

Reply:
I added this check because the existing code in inventory.go sets the host as a bootstrap after invoking `PrepareForInstallation()`
https://github.com/openshift/assisted-service/blob/a09c64feae8ac474a9cb5fb6ef2baef762b726e2/internal/bminventory/inventory.go#L1213-L1219
Also, isn't that the cluster object passed to PrepareForInstallation makes sure there are no disabled hosts?
https://github.com/openshift/assisted-service/blob/a09c64feae8ac474a9cb5fb6ef2baef762b726e2/internal/bminventory/inventory.go#L1199